.github/workflows/yocto-build-deploy.yml

```diff
@@ -967,7 +986,7 @@ jobs:
   test:
     name: Test
     needs: build
-    runs-on: ${{ fromJSON(inputs.runs-on) }}
+    runs-on: ${{ fromJSON(matrix.runs_on || '["ubuntu-22.04"]') }}
```

Comment:
I would rather rename `inputs.runs_on` to `inputs.build_runs_on` to avoid confusion, as I think it's pretty clear what a `runs_on` property inside a test matrix will do.

I would have used `inputs.runs_on` for the default value of tests runs-on, but we don't actually want that in 90% of cases right now. With the exception of QEMU tests, we would actually prefer that the tests run on a low-powered free runner so it's not using our resources, while the build runs on the biggest runner we have.

Reply:
And yes, the exclude syntax would work just as well as the include. This change gives the caller full control over the matrix as described in the GitHub docs, as long as they can convert yaml to json.

Reply:
Agreed, thanks for your first explanation. This way I think the test runner human has flexibility do whatever they want to do in the matrix as well as the combinatorial if they want. 

`build-runs-on` also sounds good.